src/DocumentFormat.OpenXml/Packaging/OpenXmlPartExtensions.cs

```diff
@@ -0,0 +1,424 @@
+
+using DocumentFormat.OpenXml.Features;
+using System;
+
+namespace DocumentFormat.OpenXml.Packaging
+{
+    /// <summary>
+    /// Defines extensions for part relationships
+    /// </summary>
+    public static class OpenXmlPartExtensions
+    {
+        private static TChild InitPart<TChild>(this OpenXmlPartContainer parent, TChild childPart, PartTypeInfo partType, string? relId = null)
+            where TChild : OpenXmlPart
+        {
+            if (parent == null)
+            {
+                throw new ArgumentNullException(paramName: nameof(parent));
+            }
+
+            string contentType = partType.ContentType;
+
+            var partExtension = partType.Extension;
+            parent.Features.GetRequired<IPartExtensionFeature>().Register(contentType, partExtension);
+
+            if (relId == null || relId.Length == 0)
+            {
+                relId = null;
+            }
+
+            parent.InitPart(childPart, contentType, relId);
+            return childPart;
+        }
+
+        /// <summary>
+        /// Adds a AlternativeFormatImportPart as a relationship to the parent part
+        /// </summary>
+        /// <param name="parent">The parent part requesting to add.</param>
+        /// <param name="partType">The part type information for the added extensible part.</param>
+        /// <param name="id">The relationship id. Optional, default to null.</param>
+        /// <return>The newly added part</return>
+        /// <see cref="AlternativeFormatImportPart"/>
+        public static AlternativeFormatImportPart AddAlternativeFormatImportPart<T>(this T parent, PartTypeInfo partType, string? id = null)
+            where T : OpenXmlPart, ISupportRelationship<AlternativeFormatImportPart> => OpenXmlPartExtensions.InitPart<AlternativeFormatImportPart>(parent, new AlternativeFormatImportPart(), partType, id);
+
+        /// <summary>
+        /// Adds a AlternativeFormatImportPart as a relationship to the parent part
+        /// </summary>
+        /// <param name="parent">The parent part requesting to add.</param>
+        /// <param name="contentType">The content type information for the added extensible part.</param>
+        /// <param name="id">The relationship id. Optional, default to null.</param>
+        /// <return>The newly added part</return>
+        /// <see cref="AlternativeFormatImportPart"/>
+        public static AlternativeFormatImportPart AddAlternativeFormatImportPart<T>(this T parent, string contentType, string? id = null)
+            where T : OpenXmlPart, ISupportRelationship<AlternativeFormatImportPart>
+        {
+            AlternativeFormatImportPart childPart = new();
+
+            if (parent == null)
+            {
+                throw new ArgumentNullException(nameof(parent));
+            }
+
+            parent.InitPart(childPart, contentType, id);
+            return childPart;
```

Comment:
```suggestion
            if (parent == null)
            {
                throw new ArgumentNullException(nameof(parent));
            }

            return parent.InitPart(new AlternativeFormatImportPart(), contentType, id);
```

And so on for the others

Reply:
parent doesn't return a part, that's why I added the childPart local.

Reply:
Just saw this. You can add a helper method here that does that in this class. It makes a lot more verbose